proxmox/client.go

```diff
@@ -2213,3 +2226,59 @@ func (c *Client) CheckTask(resp *http.Response) (exitStatus string, err error) {
 	}
 	return c.WaitForCompletion(taskResponse)
 }
+
+// Returns the Client's cached version if it exists, otherwise fetches the version from the API.
+func (c *Client) Version() (Version, error) {
+	if c == nil {
+		return Version{}, errors.New(Client_Error_Nil)
+	}
+	if c.version == nil {
+		return c.GetVersion()
+	}
+	c.versionMutex.Lock()
+	defer c.versionMutex.Unlock()
+	return Version{
+		Major: c.version.Major,
+		Minor: c.version.Minor,
+		Patch: c.version.Patch,
+	}, nil
+}
+
+type Version struct {
+	Major uint8
+	Minor uint8
+	Patch uint8
+}
+
+// Greater returns true if the version is greater than the other version.
+func (v Version) Greater(other Version) bool {
+	return uint32(v.Major)*256*256+uint32(v.Minor)*256+uint32(v.Patch) > uint32(other.Major)*256*256+uint32(other.Minor)*256+uint32(other.Patch)
```

Comment:
This is instead of multiple consecutive `if` statements. It needed a unit test anyways so might as well implement the most performant solution.